src/proposal.mligo

```diff
@@ -344,27 +348,32 @@ let handle_proposal_is_over
     )
     : (operation list * storage * counter) =
   let proposal = check_if_proposal_exist (proposal_key, store) in
-  if    is_voting_period_over (proposal, config.voting_period, store)
-     && counter.current < counter.total // not finished
-  then
-    let counter = { counter with current = counter.current + 1n } in
-    let cond =    do_total_vote_meet_quorum_threshold(proposal, store, config.quorum_threshold)
-               && proposal.upvotes > proposal.downvotes
-    in
-    let store = unfreeze_proposer_and_voter_token
-          (config.rejected_proposal_return_value, cond, proposal, config.voting_period, config.fixed_proposal_fee_in_token, store) in
-    let (new_ops, store) =
-      if cond
-      then
-        let (ops, new_extra) = config.decision_lambda (proposal, store.extra)
-        in (ops, { store with extra = new_extra })
-      else (([] : operation list), store)
-    in
-    let cons = fun (l, e : operation list * operation) -> e :: l in
-    let ops = List.fold cons ops new_ops in
-    let store = delete_proposal (start_date, proposal_key, store) in
-    (ops, store, counter)
-  else (ops, store, counter)
+
+  if is_time_reached (proposal, config.proposal_expired_time)
+    then (failwith("EXPIRED_PROPOSAL") : (operation list * storage * counter))
```

Comment:
`EXPIRED_PROPOSAL` should be added to the error list in the spec.

Reply:
I've added this in a separate commit: [`4669c84` (#232)](https://github.com/tqtezos/baseDAO/pull/232/commits/4669c84ecec2ec93723844ae16bab7902ae9c0a3)

Have to do this, to avoid problem with rebasing.